internal/controller/streams.go

```diff
@@ -560,7 +562,10 @@ func ConnectStreams(ctx context.Context, logger *zap.Logger, envMap EnvironmentC
 				ciliumNamespace:           envMap.CiliumNamespace,
 				disableNetworkFlowsCilium: false,
 				falcoEventChan:            falcoEventChan,
+				flowCollector:             pb.FlowCollector_FLOW_COLLECTOR_UNSPECIFIED,
 			}
+			// Initialize the WaitGroup with a count of 1.
+			streamClient.flowCollectorWG.Add(1)
```

Comment:
I'm sorry, I gave you the wrong feedback.
We don't need synchronization / a WaitGroup. We're already doing things in the right order.
```suggestion
```